src/pages/Search/SearchPageBottomTab.tsx

```diff
@@ -56,30 +96,51 @@ function SearchPageBottomTab() {
                 shouldShowLink={false}
             >
                 {!selectionMode?.isEnabled && queryJSON ? (
+                    <Animated.View style={animatedHeaderStyle}>
+                        <View style={styles.appBG}>
+                            <TopBar
+                                activeWorkspaceID={policyID}
+                                breadcrumbLabel={translate('common.search')}
+                                shouldDisplaySearch={false}
+                            />
+                        </View>
+                        <Animated.View style={animatedTopBarStyle}>
+                            <SearchTypeMenu
+                                isCustomQuery={isCustomQuery}
+                                queryJSON={queryJSON}
+                            />
+                            {shouldUseNarrowLayout && (
+                                <SearchStatusBar
+                                    type={queryJSON.type}
+                                    status={queryJSON.status}
+                                />
+                            )}
+                        </Animated.View>
+                    </Animated.View>
+                ) : (
                     <>
-                        <TopBar
-                            activeWorkspaceID={policyID}
-                            breadcrumbLabel={translate('common.search')}
-                            shouldDisplaySearch={false}
-                        />
-                        <SearchTypeMenu
-                            isCustomQuery={isCustomQuery}
-                            queryJSON={queryJSON}
+                        <HeaderWithBackButton
+                            title={translate('common.selectMultiple')}
+                            onBackButtonPress={() => {
+                                clearSelectedTransactions();
+                                turnOffMobileSelectionMode();
+                            }}
                         />
+
+                        {queryJSON && (
+                            <SearchPageHeader
+                                queryJSON={queryJSON}
+                                hash={queryJSON.hash}
+                                isCustomQuery={isCustomQuery}
+                            />
+                        )}
```

Comment:
Do you think its possible to somehow simplify this and reduce the amount of conditional rendering inside this return?
We have conditional rendering for `TopBar` + `SearchTypeMenu`, then extra condition for `SearchStatusBar` and all that is exclusive with `SearchPageHeader` rendering.
It's very hard to understand which parts of the UI will be rendered.

Some ideas:
 - early returns?
 - saving parts of JSX to variables?
 - small helper components declared within this file?
 
I think you could try to go over this code again and see if anything can be improved.